sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -436,6 +436,9 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
       .replaceAll(
         s"Location.*/sql/core/spark-warehouse/$clsName/",
         s"Location ${notIncludedMsg}sql/core/spark-warehouse/")
+      .replaceAll(
+        s"Location.*\\.\\.\\.",
```

Comment:
shall we merge it with the previous case and always normalize the location string?

Reply:
@cloud-fan Actually i had thought about it. But for some tests, i wasn't sure if we should change the output. For example : 
```sql/core/src/test/resources/sql-tests/results/describe-part-after-analyze.sql```

the test is : 
DESC EXTENDED t PARTITION (ds='2017-08-01', hr=10)

Perhaps displaying the location is the intention. What do you think ?

Reply:
If we decide to do it, i feel we should do it in another pr. So if required, can be reverted easily ?

Reply:
oh wait, why does the previous case doesn't work for the new changes?

Reply:
@cloud-fan because, we truncate the location to 100 chars for explain.

Reply:
do you mean the `sql/core/spark-warehouse` part will be truncated?

Reply:
@cloud-fan Yeah Wenchen. Depending on the path truncation may happen ? But when it happens the end pattern becomes "...." and thats what i am looking for here. Let me know please ..
```
val abbreviatedLoaction = StringUtils.abbreviate(redact(value), 100)
```

Reply:
do we mean the table location is not truncated in the old explain? Otherwise I don't understand why the previous case doesn't work only for the table location in the new explain.

Reply:
@cloud-fan So we have only started to test the explain formatted feature using SQLQueryTestSuite infrastructure. Thats why we didn't see any issue with truncation w.r.t explain.

Reply:
How about we don't truncate the location in the new explain?

Reply:
This string replace is initially introduced at https://github.com/apache/spark/pull/16373, for SHOW TABLE which doesn't truncate the location property.

Reply:
> How about we don't truncate the location in the new explain?

Running `test("explain formatted - check presence of subquery in case of DPP") ` in ExplainSuite

Wenchen, i am afraid, the location can be arbitrarily large like following. Are we sure we want to show everything ? It may make the plan look un-readable again ?

```
PrunedInMemoryFileIndex[file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=193, 
file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=546, 
file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=0, 
file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=27, 
file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=572, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=333, 
file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=451, 
file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=56, 
file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=55, 
file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
warehouse/org.apache.spark.sql.ExplainSuite/df1/k=628, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=937, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=609, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=37, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=7, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=908, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=34, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=621, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=596, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=108, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=42, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=990, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=294, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=418, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=749, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-warehouse/org.apache.spark.sql.ExplainSuite/df1/k=606, file:/Users/dilipbiswal/mygit/apache/spark/sql/core/spark-
....
....
....
..... 
.....
```

Reply:
ah, this is very different from SHOW TABLE.

We need to think about how to display a list of locations in the explain output. I think it's more useful to display only the first location, with the number of remaining locations if there are any.

Reply:
@cloud-fan OK.. makes sense.